packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1091,6 +1108,29 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
 
       final double xOffset = hasCustomWidth ? (size.width - snackBarWidth!) / 2 : 0.0;
       positionChild(_ScaffoldSlot.snackBar, Offset(xOffset, snackBarYOffsetBase - snackBarSize.height));
+
+      assert((){
+        // Whether a floating SnackBar has been offsetted too high.
+        // Assert after call to positionChild otherwise SnackBar position differs in debug mode.
```

Comment:
I am not sure I follow this comment, why do we have to wait until after calling positionChild? We have the information we need to assert prior to the call right?

Reply:
Hey @Piinks!
If the assert is done before calling positionChild, the error is thrown and the Snackbar is positionned at (0,0).
Here is a screenshot :
![Capture d’écran du 2022-05-12 10-31-59](https://user-images.githubusercontent.com/840911/168060473-f637159b-16fb-436c-aec5-ef5f6d7182ea.png)

If the assert is done after positionChild, the error is thrown and the SnackBar is not visible :
![Capture d’écran du 2022-05-12 10-31-23](https://user-images.githubusercontent.com/840911/168062626-30734e0d-5e06-4e8c-8a99-2dde98c7c5fd.png)

As we are throwing an error saying the SnackBar is not visible, It might feels strange for the user to see the Snackbar in this case. Maybe my comment could be reworded?

Reply:
Yes it does seem to be a contradiction - why would it throw in the case of the first image?

Reply:
I pushed an update where I tried to make the comment clearer :sweat_smile: